designs/Consolidate-NuGetConfig_OnLinux.md

```diff
@@ -0,0 +1,216 @@
+
+After discussing in NuGet team and checking with Visual Studio Mac/Mono team, we believe we should change the dotnet CLI path to match that of Mono’s, for the following reason:
+ * On Linux/Mac, users usually get application data from environment variable XDG_DATA_HOME, which has the default value of ~/.config. Getting application data from ~/.nuget will cause inconsistency user experience between NuGet and other applications.
+
+For the sake of simplicity, we will refer to `~/.nuget/NuGet/NuGet.Config` as legacy path, and refer to `~/.config/NuGet/NuGet.Config` as consolidated path in the following.
+
+## Who are the customers
+Dotnet CLI users who work on Linux and Mac.
+
+## Goals
+Make dotnet CLI user-wide config file path consistent with Mono on Linux/Mac and break as less dotnet CLI users as possible.
+
+## Non-Goals
+Completely get rid of the legacy path.
+
+## Background knowledge
+On Linux/Mac, user wide NuGet.Config file is created when trying to get the user wide NuGet.Config file at the beginning phase of running `dotnet restore` command(and many other dotnet commands with implicit restore if not specifying `--no-restore` option, like `dotnet new`, `dotnet build`, `dotnet run`, `dotnet test`, `dotnet publish`, and `dotnet pack`), or `dotnet msbuild -t:restore` command, but the user wide NuGet.Config file does not exist. Then the user wide NuGet.Config will be created with package source of nuget.org, as the only default package source. 
+
+## Potential Solutions overview 
+### Solution 1: Use only consolidated path
+Change NuGet to use the consolidated config path for dotnet cli. Customers using only new tooling have no reason to keep the legacy path.
+
+#### Implementation
+Add following steps when trying to get user-wide config file path by running dotnet CLI on Linux/Mac:
+Check if the following two conditions are satisfied.
+* The legacy path does not exist.
+*	The consolidated path exists.
+
+If the two conditions are satisfied, NuGet will get user wide NuGet.Config path from consolidated path.
+
+If not, it should be one of the following cases:
+
+**1. Neither legacy path nor consolidated exists.**
+
+NuGet will create a default NuGet.Config file in consolidated path.
+NuGet will get user wide NuGet.Config file from consolidated path. 
+No warning will be shown.
+
+**2. Legacy path exist, but consolidated path does not exist.**
+
+NuGet will copy the NuGet.Config file from legacy path to consolidated path, then delete the one in legacy path. 
+NuGet will get user wide NuGet.Config file from consolidated path.
+No warning will be shown.
```

Comment:
I think an informational message should be displayed is we're moving the config file from a path that older tools use to a path that older tools will not use.  The message might get lost if it's in a long restore log. But silently moving files that might affect other tools doesn't feel like a great idea to me.